pre_commit/staged_files_only.py

```diff
@@ -57,7 +58,14 @@ def _unstaged_changes_cleared(patch_dir: str) -> Generator[None, None, None]:
             patch_file.write(diff_stdout_binary)
 
         # Clear the working directory of unstaged changes
-        cmd_output_b('git', 'checkout', '--', '.')
+        #
+        # Do not use "git checkout" to avoid triggering post-checkout hooks.
+        #
+        #       https://github.com/pre-commit/pre-commit/issues/1418
+        #
```

Comment:
the whitespace and formatting is super unnecessary here.  this should be at most one line, and you can refer to an issue using its number like `#1418`

`cmd_output` / `cmd_output_b` take an `env` environment dictionary which would be better I think than `envcontext` here

Reply:
Sorry, didn't you say use `envcontext` above?

I can change it to use a dict instead, and I've made the comment one line.

Reply:
I indicated that your before/after thing is better done with `envcontext`, but that in this case an env dictionary is best